lib/elixir/lib/inspect/algebra.ex

```diff
@@ -1019,17 +1033,20 @@ defmodule Inspect.Algebra do
   #
   #   * flat - represents a document with breaks as flats (a break may fit, as it may break)
   #   * break - represents a document with breaks as breaks (a break always fits, since it breaks)
+  #
+  # These other two modes only affect fitting:
+  #
   #   * flat_no_break - represents a document with breaks as flat not allowed to enter in break mode
   #   * break_no_flat - represents a document with breaks as breaks not allowed to enter in flat mode
   #
   @typep mode :: :flat | :flat_no_break | :break | :break_no_flat
 
-  @spec fits(
+  @spec fits?(
```

Comment:
It can now be a boolean because `format` injects a fake `:group_over` document that allows us to track where the parent groups stops. And by making `next_break_fits`, we can avoid look aheads.

Reply:
Since this changes the contract between the formatter and the algebra, is there a risk for this to be a breaking change for libraries relying on algebra? (e.g. now they need to add a fake `|> group()` too?

Reply:
I have added doc_fits back and I confirmed that the formatter without the changes in this PR still works, except for the `next_break_fits` bug you fixed (which caused this function to change). So if they want the fix, they need to move to the new API. :)

Reply:
I think that's fair, it was quite the edge-case bug 🙂